oak/server/oak_node.cc

```diff
@@ -187,29 +207,30 @@ std::unique_ptr<OakNode> OakNode::Create(const std::string& module) {
     return nullptr;
   }
 
-  wabt::interp::Thread::Options thread_options;
-  // wabt::Stream* trace_stream = s_stdout_stream.get();
-  wabt::Stream* trace_stream = nullptr;
-  wabt::interp::Executor executor(&node->env_, trace_stream, thread_options);
-  LOG(INFO) << "Executing module";
-
-  // Create a (persistent) logging channel to allow the module to log during initialization.
-  node->channel_halves_[LOGGING_CHANNEL_HANDLE] = absl::make_unique<LoggingChannelHalf>();
-  LOG(INFO) << "Created logging channel " << LOGGING_CHANNEL_HANDLE;
-
-  wabt::interp::TypedValues args;
-  wabt::interp::ExecResult exec_result =
-      executor.RunExportByName(node->module_, "oak_initialize", args);
-
-  if (exec_result.result != wabt::interp::Result::Ok) {
-    LOG(WARNING) << "Could not execute module initialization";
-    wabt::interp::WriteResult(s_stdout_stream.get(), "error", exec_result.result);
-    // TODO: Print error.
-    return nullptr;
+  // Create a logging channel for the module.
+  {
+    std::shared_ptr<MessageChannel> channel = std::make_shared<MessageChannel>();
+    node->channel_halves_[LOGGING_CHANNEL_HANDLE] =
+        absl::make_unique<MessageChannelWriteHalf>(channel);
+    LOG(INFO) << "Created logging channel " << LOGGING_CHANNEL_HANDLE;
+
+    // Spawn a thread that reads and logs messages on this channel forever.
+    std::thread t([channel] {
```

Comment:
I suppose we need to think how to reclaim / garbage collect this at some point. Perhaps storing it as a member of `OakNode` would help?

Reply:
Yeah, there's definitely a Node lifetime issue we need to think about.  I've dropped some TODOs in for now.